### ISSUE TITLE
[Bug]: TestComponent is not a <Route> component, even when it is

### ISSUE THREAD

**Issue by pablo-mayrgundter** [remix-run/react-router#8343]
### What version of React Router are you using?

6.0.2

### Steps to Reproduce

```
const TestRoute = () => <Route path="b" element={<div>Child</div>}/>
....
<Routes>
  <Route
    path="/a/*"
    element={<div>Parent</div>}>
    <TestRoute/>
  </Route>
</Routes>
```
### Expected Behavior

The TestRoute is just a rename of Route component and should be allowed.  The type check seems too thorough.. maybe a factory method that allows creation of Route.

Without this, it's not clear to me that it's possible to do recursive Path defintions.

### Actual Behavior

I can't import a Route tag defined in another component.

**Comment by timdorr** [comment]
https://github.com/remix-run/react-router/issues/8294#issuecomment-964187831

**Comment by pablo-mayrgundter** [comment]
Is it right then that there's no way to do recursive JSX route composition in v6?

It doesn't look like this is addressed in #8294.

e.g. smth like:

const NavTree = (depth) =>
  <Route path={depth} element={<div>{depth}</div>}>
    depth > 0 ? <NavTree depth={depth - 1} /> : null
  </Route>

....
<Routes>
  <Route path="/tree/*" element={<div>Nav Tree</div>}>
    <NavTree depth={3}/>
  </Route>
</Routes>
